### DISCUSSION TITLE
step to install into laravel breeze using npm

### DISCUSSION THREAD

**Discussion by idiey** [moesaid/cleopatra#13] [General]
Hi,

i need guide on how to use this package after install using npm into laravel 8/laravel breeze.

Thanks

**Comment by NikarashiHatsu** [comment]
Hello, let me help you through your problem based on my understanding.
I don't really know how to publish this package to Laravel instantly, but try the following:

1. Clone the repository `git clone https://github.com/moesaid/cleopatra.git`
2. Run `npm install`
3. Then, run `npm run build`. This command will compile the css and js, in the future, those files will be stored on a folder called `build`
4. Open the `build` folder, and you will see 3 folders, if you're using laravel, copy those folders to your Laravel's public folder
5. You can copy the rest of the HTML to your Laravel's `resources/views` folder. Convert those to `.blade.php` files

Best regards.

**Reply by turdakhov** [reply]
Hi! Have an error on "npm run build" :(

daur@DESKTOP-U9CBLMQ:~/discover$ sail npm run build
npm ERR! Missing script: "build"
npm ERR!
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

**Comment by moesaid** [comment]
@idiey this is not a laravel package, never less you can use it with any web framework as the package compile to vanilla css&js, you can fallow @NikarashiHatsu instruction.

**Comment by Geovanek** [comment]
I tried via via npm
npm install @moesaid/cleopatra

So I ran npm install && npm run dev

Then I put it on webpack.mix
```
mix.js ('resources / js / app.js', 'public / js'). postCss ('resources / css / app.css', 'public / css', [
     require ('postcss-import'),
     require ('tailwindcss'),
     require ('autoprefixer'),
     require ('@ moesaid / cleopatra'),
]);
```

But that returned me with several errors with the gulp.

```
[webpack-cli] Error: Cannot find module 'gulp'
Require stack:
- /var/www/bikeparkgk/node_modules/@moesaid/cleopatra/gulpfile.js
- /var/www/bikeparkgk/webpack.mix.js
- /var/www/bikeparkgk/node_modules/laravel-mix/setup/webpack.config.js
- /var/www/bikeparkgk/node_modules/webpack-cli/lib/webpack-cli.js
- /var/www/bikeparkgk/node_modules/webpack-cli/lib/bootstrap.js
- /var/www/bikeparkgk/node_modules/webpack-cli/bin/cli.js
- /var/www/bikeparkgk/node_modules/webpack/bin/webpack.js
    at Function.Module._resolveFilename (node: internal / modules / cjs / loader: 941: 15)
    at Function.Module._load (node: internal / modules / cjs / loader: 774: 27)
    at Module.require (node: internal / modules / cjs / loader: 1013: 19)
    at require (/var/www/bikeparkgk/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object. <anonymous> (/var/www/bikeparkgk/node_modules/@moesaid/cleopatra/gulpfile.js:1:116)
    at Module._compile (/var/www/bikeparkgk/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (node: internal / modules / cjs / loader: 1138: 10)
    at Module.load (node: internal / modules / cjs / loader: 989: 32)
    at Function.Module._load (node: internal / modules / cjs / loader: 829: 14)
    at Module.require (node: internal / modules / cjs / loader: 1013: 19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/var/www/bikeparkgk/node_modules/@moesaid/cleopatra/gulpfile.js',
    '/var/www/bikeparkgk/webpack.mix.js',
    '/var/www/bikeparkgk/node_modules/laravel-mix/setup/webpack.config.js',
    '/var/www/bikeparkgk/node_modules/webpack-cli/lib/webpack-cli.js',
    '/var/www/bikeparkgk/node_modules/webpack-cli/lib/bootstrap.js',
    '/var/www/bikeparkgk/node_modules/webpack-cli/bin/cli.js',
    '/var/www/bikeparkgk/node_modules/webpack/bin/webpack.js'
  ]
}
```

I will try via GIT as presented above

**Comment by ghost** [comment]
checkout https://github.com/digiqrp/Cleopatra it's a Laravel 9 project based on this excellent work

**Reply by rstriquer** [reply]
This digigrp repo seems to be off (unfortunately) ...

![image](https://user-images.githubusercontent.com/669705/228947674-4183de5c-7396-4456-b2b9-bcc563fb86ab.png)

**Comment by rstriquer** [comment]
I had to install the following packages: ```gulp```; ```browser-sync```; ```gulp-sass```; ```gulp-concat```; ```gulp-postcss```; ```gulp-purifycss```; ```gulp-uglify```; and ```gulp-imagemin```. I tried to use Node V19 and V16, but in all scenarios, I received the same error in my Laravel project.

![image](https://user-images.githubusercontent.com/669705/228979504-636bbbc8-080b-4026-bd5d-d7840b0dcd49.png)
